api/app/Http/Controllers/AuthController.php

```diff
@@ -245,22 +219,12 @@ public function handleProviderCallback($provider, Socialite $socialite, User $us
      * @param  string  $requester
      * @param  Request $request
      *
-     * @return Response
+     * @return ApiResponse
      */
     public function singleSignOn($requester, Request $request)
     {
-        // A single sign on request might have different requirements and
-        // methods how to deal with a non logged in user. So we get the user
-        // if possible, and if not we pass in a null user and let the the
-        // requester class deal with it according to the requester's definitions
-        if ($token = $request->cookie(self::JWT_AUTH_TOKEN_COOKIE)) {
-            $user = $this->jwtAuth->toUser($token);
-        } else {
-            $user = null;
-        }
-
+        $user = $this->auth->user();
```

Comment:
Will the auth service also check the cookies? Is that why this works? The user calling this function won't actually be passing the Authorization header (they're coming from the vanilla forums).